src/shared/XLinkPrivateFields.c

```diff
@@ -87,7 +89,9 @@ streamDesc_t* getStreamByName(xLinkDesc_t* link, const char* name)
         if (link->availableStreams[stream].id != INVALID_STREAM_ID &&
             strcmp(link->availableStreams[stream].name, name) == 0) {
             XLink_sem_wait(&link->availableStreams[stream].sem);
-
+            if(link->availableStreams[stream].id == INVALID_STREAM_ID){
```

Comment:
I also don't (yet) see the value of this addition lines 92-94. The check for INVALID_STREAM_ID is done on line 89.